### PR TITLE
exui-2261-globalsearch-service-name-list

### PR DESCRIPTION
**Pull request by RiteshHMCTS**
### Jira link
https://tools.hmcts.net/jira/browse/EXUI-2261

### Change description
globalsearch-service-name-list

### CODE REVIEW

**Review by anthonydummer — APPROVED** [review]
LGTM

**Review by MunishSharmaHMCTS — APPROVED** [review]

File: api/globalSearch/index.ts [hunk 1]
```diff
@@ -53,17 +60,17 @@ export async function getSearchResults(req: EnhancedRequest, res: Response, next
...
   const globalSearchServiceIdsArray = globalSearchServiceIds.split(',');
 
   // Generate global search services
   const globalSearchServices: GlobalSearchService[] = [];
   globalSearchServiceIdsArray.forEach((serviceId) => {
-    const jurisdiction = jurisdictions ? jurisdictions.find((x) => x.id === serviceId) : null;
+    const jurisdiction = refDataHMCTS ? refDataHMCTS.find((x) => x.ccd_service_name === serviceId) : null;
     if (jurisdiction) {
-      globalSearchServices.push({ serviceId: jurisdiction.id, serviceName: jurisdiction.name });
+      globalSearchServices.push({ serviceId: jurisdiction.ccd_service_name, serviceName: jurisdiction.ccd_service_name });
```
Comment (andywilkinshmcts) [inline api/globalSearch/index.ts:73]: service name should be set to "jurisdiction.service_short_description"
Need to check that the from end code uses the serviceName field to populate the drop down (not the service Id)

**Review by andywilkinshmcts — CHANGES_REQUESTED** [review]

File: api/globalSearch/index.ts (hunk 1 above)
Comment (RiteshHMCTS) [reply api/globalSearch/index.ts]: I have made the changes as mentioned above

**Review by MunishSharmaHMCTS — APPROVED** [review]

File: api/globalSearch/index.spec.ts [hunk 2]
```diff
@@ -9,26 +8,142 @@ import { mockReq, mockRes } from 'sinon-express-mock';
...
+      sub_business_area: 'Civil and Family',
+      service_description: 'Family Public Law',
+      service_code: 'ABA3',
+      service_short_description: 'Family Public Law',
+      ccd_service_name: 'PUBLICLAW',
+      last_update: '2021-02-10T11:53:58.808063',
+      ccd_case_types: ['CARE_SUPERVISION_EPO']
+    },
+    {
+      jurisdiction: 'Employment Tribunerals',
```
Comment (andywilkinshmcts) [inline api/globalSearch/index.spec.ts:89]: Not important, but typo in Employment Tribunals

File: api/globalSearch/index.ts [hunk 3]
```diff
@@ -53,17 +60,19 @@ export async function getSearchResults(req: EnhancedRequest, res: Response, next
...
   // Generate global search services
   const globalSearchServices: GlobalSearchService[] = [];
   globalSearchServiceIdsArray.forEach((serviceId) => {
-    const jurisdiction = jurisdictions ? jurisdictions.find((x) => x.id === serviceId) : null;
+    // search for the service name based on the globalSearchServiceId
+    const jurisdiction = refDataHMCTS?.length > 0 ? refDataHMCTS.filter((x) => x.ccd_service_name?.toLowerCase() === serviceId.toLowerCase()) : null;
     if (jurisdiction) {
-      globalSearchServices.push({ serviceId: jurisdiction.id, serviceName: jurisdiction.name });
+      // handle Civil service which has different service_short_description
+      globalSearchServices.push({ serviceId: jurisdiction[0].ccd_service_name, serviceName: jurisdiction.length > 1 ? 'Civil' : jurisdiction[0].service_short_description });
```
Comment (andywilkinshmcts) [inline api/globalSearch/index.ts:75]: Not quite what I wanted, but close.
I think what I wanted was
if (jurisdiction.length > 1) {
   globalSearchServices.push({ serviceId: jurisdiction[0].ccd_service_name, serviceName: jurisdiction[0].toTitleCase(jurisdiction[0].serviceName));
} else {
   globalSearchServices.push({ serviceId: jurisdiction[0].ccd_service_name, serviceName: jurisdiction[0].service_short_description);

}
function toTitleCase(str: string): string {
str.toLowerCase().replace(/\b\w/g, s => s.toUpperCase());

**Review by andywilkinshmcts — CHANGES_REQUESTED** [review]

File: api/globalSearch/index.spec.ts (hunk 2 above)
Comment (RiteshHMCTS) [reply api/globalSearch/index.spec.ts]: Changed

File: api/globalSearch/index.ts (hunk 3 above)
Comment (RiteshHMCTS) [reply api/globalSearch/index.ts]: I have implemented as above but the above regex wont work with all letter with CAPITAL so changed accoringly so it should convert from text from CAPITAL to Capital

File: api/globalSearch/index.ts [hunk 4]
```diff
@@ -72,3 +85,9 @@ export function generateServices(jurisdictions: Jurisdiction[]): GlobalSearchSer
   // Return generated global search services
   return globalSearchServices;
 }
+
+function toTitleCase(serviceName: string): string {
+  return serviceName.replace(/([a-zA-Z])([a-zA-Z]*)/g, (match, firstLetter, rest) => {
+    return firstLetter.toUpperCase() + rest.toLowerCase();
```
Comment (andywilkinshmcts) [inline api/globalSearch/index.ts:91]: That won't work for strings that have non alpha chars, such as "andy_wilkins", but I guess it doesn't matter

**Review by andywilkinshmcts — APPROVED** [review]

**Pull request merged**